core/services/chainlink/application.go

```diff
@@ -221,25 +227,12 @@ func NewApplication(config *orm.Config, ethClient eth.Client, advisoryLocker pos
 		subservices: subservices,
 	}
 
-	headTrackables := []strpkg.HeadTrackable{gasUpdater}
-
-	headTrackables = append(
-		headTrackables,
-		ethConfirmer,
-		jobSubscriber,
-		pendingConnectionResumer,
-		balanceMonitor,
-		promReporter,
-		logBroadcaster,
-	)
-
-	for _, onConnectCallback := range onConnectCallbacks {
-		headTrackable := &headTrackableCallback{func() {
+	go func() {
+		appCallbackLock.Lock()
+		for _, onConnectCallback := range onConnectCallbacks {
```

Comment:
To be honest I think we should remove the OnConnect/OnDisconnect callbacks entirely if possible. Most headtracker clients dont need them

Reply:
TBH I'm all for it also. I can't say definitively that no headtracker services use the onconnect / disconnect function, but from my brief glance I didn't see much functionality. That refactor might need to be a subsequent PR though since I'm just trying to get unblocked on the keeper work.